src/librustc_errors/emitter.rs

```diff
@@ -1629,6 +1629,19 @@ impl EmitterWriter {
                             _ => ()
                         }
                     }
+                    if children.is_empty() && suggestions.iter().any(|s| {
+                        s.style != SuggestionStyle::CompletelyHidden
+                    }) {
+                        let mut buffer = StyledBuffer::new();
```

Comment:
This logic seems identical (copied?) from above; can you de-dup?

Reply:
done